lib/query-builders/applicant.js

```diff
@@ -0,0 +1,24 @@
+const {
+  autoResolved,
+  returnedToApplicant,
+  resolved,
+  rejected,
+  withdrawn
+} = require('../flow/status');
+
+module.exports = {
+  outstanding: (queryBuilder, profile) => {
+    queryBuilder.whereJsonSupersetOf('data', { subject: profile.id })
+      .andWhere('status', returnedToApplicant.id);
+  },
+
+  inProgress: (queryBuilder, profile) => {
+    queryBuilder.whereJsonSupersetOf('data', { subject: profile.id })
+      .whereNotIn('status', [autoResolved.id, resolved.id, rejected.id, withdrawn.id, returnedToApplicant.id]);
```

Comment:
The status strings in the db aren't quite identical to the name of the object (e.g. `autoResolved` vs `autoresolved` and `referredToInspector` vs `referred-to-inspector`), and the column is a string so the `whereNotIn()` needs the strings rather than the objects.

The reason for using the status objects everywhere is if there's a typo it won't actually run, rather than just silently being a bug as would happen with a hardcoded string.

We could write a function to return the id but then you'd have to call the function for each object so I'm not sure that's an improvement?

Reply:
yeah as you were then, as below just a comment but if there's a valid reason for it then that's fine by me 👍